### PR TITLE
Convert hashicorp/waypoint-hzn to GitHub Actions

### PR DESCRIPTION
**Pull request by hc-github-team-es-release-engineering**
Pipeline migrated from [CircleCI](https://app.circleci.com/pipelines/github/hashicorp/waypoint-hzn) :tada:

## Manual steps

Perform the following steps to complete the migration:

### hashicorp/waypoint-hzn/go-tests
- [x] Ensure secret is available: `${{ secrets.CIRCLE_CI_SSH_KEY }}`
- [x] Ensure secret is available: `${{ secrets.CIRCLE_CI_KNOWN_HOSTS }}`
- [x] Ensure environment variable is updated: `GH_DOCKER_USER`

### hashicorp/waypoint-hzn/integration
- [x] Ensure secret is available: `${{ secrets.CIRCLE_CI_SSH_KEY }}`
- [x] Ensure secret is available: `${{ secrets.CIRCLE_CI_KNOWN_HOSTS }}`
- [x] Ensure environment variable is updated: `GH_DOCKER_USER`

### CODE REVIEW

File: .github/workflows/go-tests.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          go-version-file: go.mod
+      - id: go-version
+        run: echo "go-version=$(cat ./.go-version)" >> "$GITHUB_OUTPUT"
+      - uses: hashicorp/actions-go-build@v0.1.7
+        with:
+          go_version: ${{ steps.go-version.outputs.go-version }}
+          os: linux
+          arch: amd64
+          reproducible: nope
+          # TODO: inquire about versioning
```
Comment (emilymianeil) [inline .github/workflows/go-tests.yml:54]: Do we need any versioning at all?

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: hashicorp/waypoint-hzn/integration
+on:
+  push:
+    # TODO: do we want to push a new image everytime we push to main (including this new PR)?
+    branches:
+      # - main
```
Comment (emilymianeil) [inline .github/workflows/integration.yml:6]: Confirming that we want a new image to be pushed on every push to main?

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      - uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          path: waypoint-hzn
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      # TODO: uncomment this block to enable pushes to ghcr
```
Comment (emilymianeil) [inline .github/workflows/integration.yml:46]: Note that if we enable a new image to be released on every push to main, we this Circle CI -> GHA migration will initiate a new release.

File: .github/workflows/go-tests.yml (hunk 1 above)
Comment (catsby) [reply .github/workflows/go-tests.yml:54]: No

File: .github/workflows/integration.yml (hunk 2 above)
Comment (catsby) [reply .github/workflows/integration.yml:6]: I imagine we just remove `integration.yml` here

**Review by smacfarlane — APPROVED** [review]

**Review by catsby — APPROVED** [review]
LGTM Thank you!

**Pull request merged**